dashboard/app/reporting_external.go

```diff
@@ -19,7 +19,7 @@ import (
 // The external system is meant to poll for new bugs with apiReportingPoll,
 // and report back bug status updates with apiReportingUpdate.
 
-func apiReportingPollBugs(c context.Context, r *http.Request, payload io.Reader) (interface{}, error) {
+func apiReportingPollBugs(c context.Context, ns string, r *http.Request, payload io.Reader) (interface{}, error) {
```

Comment:
This becomes a bit ugly with 2 unused args.
One common way to do this in Go is middleware wrappers. Namely all top handlers have `func(c context.Context, r *http.Request)` signature. Then we add `namespaceHandler` wrapper that accepts 
`func(c context.Context, ns string, r *http.Request)` handler that transforms it to `func(c context.Context, r *http.Request)`, it extracts/checs `ns` http parameter. Similarly `payloadHandler` accepts `func(c context.Context, ns string, r *http.Request, payload io.Reader)` and transforms it to namespace handler.
Then handlers don't need all unused arguments, and the table would look like:
```
	"log_error": apiLogError,
	"upload_build": namespaceHandler(apiUploadBuild),
	"save_coverage": namespaceHandler(payloadHandler(apiSaveCoverage)),
```

Reply:
Ack. I missed the fact all our handlers have "r *http.Request" and don't use it.